base_tier_validation_helper/models/tier_definition_helper.py

```diff
@@ -0,0 +1,230 @@
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    auto_name = fields.Boolean(
+        string='Auto name',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='Auto name for tier definition',
+    )
+    notify_on_create = fields.Boolean(
+        string="Notify Reviewers on Creation",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="If set, all possible reviewers will be notified by email when "
+             "this definition is triggered."
+    )
+    has_comment = fields.Boolean(
+        string='Comment',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    approve_sequence = fields.Boolean(
+        string='Approve by sequence',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="Approval order by the specified sequence number",
+    )
+    state = fields.Selection(
+        string='Status',
+        selection=[
+            ('draft', 'Draft'),
+            ('done', 'Created'),
+            ('cancel', 'Deleted'),
+        ],
+        default='draft',
+        index=True,
+        copy=False,
+        track_visibility='onchange',
+        required=True,
+        readonly=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name='tier.definition.helper.line',
+        inverse_name='parent_id',
+        string='Lines',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    @api.constrains('line_ids')
+    def _check_line_ids(self):
+        if self.state == 'done' and \
+                any(not line.tier_definition_id for line in self.line_ids):
+            raise UserError(_('Cannot add a line on Created state!'))
+
+    @api.onchange('model_id')
+    def _onchange_model_id(self):
+        models = self.env['tier.definition']._get_tier_validation_model_names()
+        return {'domain': {'model_id': [('model', 'in', models)]}}
+
+    @api.multi
+    def button_draft(self):
+        self.write({'state': 'draft'})
+
+    @api.multi
+    def button_create_tier(self):
+        if not self.line_ids:
+            raise ValidationError(_('Please add at least one line.'))
+        tier_definition_env = self.env['tier.definition']
+        for rec in self.line_ids:
+            if not rec.tier_definition_id:
+                tier_definition = tier_definition_env.create(
+                    rec._prepare_tier_definition())
+                rec.tier_definition_id = tier_definition
+        self.write({'state': 'done'})
+
+    @api.multi
+    def button_delete_tier(self):
+        self.line_ids.mapped('tier_definition_id').unlink()
+        self.write({'state': 'cancel'})
+
+
+class TierDefinitionHelperLine(models.Model):
+    _name = 'tier.definition.helper.line'
+    _description = 'Tier Definition Helper Line'
+
+    parent_id = fields.Many2one(
```

Comment:
Naming Convention
(parent_id normally means same object but have hierarchy)
```suggestion
    helper_id = fields.Many2one(
```

Reply:
Edited